ecdsa_adaptor.md

```diff
@@ -0,0 +1,196 @@
+  - Scalars have a postfix operation (`^1`) which denotes inversion modulo *n*.
+  - Points are members of the secp256k1 group.
+  - Points have two binary operations between them: addition (`+`) and subtraction (`-`) which represent the group operation and the inverse group operation respectively.
+  - Points and scalars have a infix multiplication (`*`) operation which for `s * P` means adding the point `P` to itself `s` times.
+  - *G* is the standard generator point of secp256k1 with coordinates:
+    - x: `0x79BE667EF9DCBBAC55A06295CE870B07029BFCDB2DCE28D959F2815B16F81798`
+    - y: `0x483ADA7726A3C4655DA4FBFC0E1108A8FD17B448A68554199C47D08FFB10D4B8`
+  - When non-zero points are encoded as bytes they are encoded using the 33-byte compact [[SEC1]] encoding (implementations should fail if they try and encode the point at infinity).
+  - When scalars are encoded as bytes they are encoded as 32-byte big endian integers.
+- other operations:
+  - `||` is an infix operation referring to byte concatenation. Its two arguments are implicitly converted to bytes.
+
+### Secure Nonce Generation
+
+Nonce generation is the most difficult aspect of cryptographic protocols to specify and the most important to get right.
+We abandon some of the responsibility for this in this document for brevity and flexibility.
+In the specification we use a function `sample_nonce` that takes a byte string and reliably returns a uniformly random scalar.
+Implementations may decide precisely how to instantiate this function.
+If `sample_nonce` is instantiated with a secure hash function then the resulting scheme is secure if we model it as a random oracle as shown by [[Bel]].
+However, we recommend adding system randomness into the process as well.
+This can be done by hashing 32 random bytes along side the input string or applying more sophisticated approach as in [[BIP340]].
+
+## Proof of Discrete Logarithm Equality
+
+As part of the ECDSA adaptor signature a proof of discrete logarithm equality must be provided.
+In our case this is a proof that the discrete logarithm of some `X` to the standard base `G` is the same as the discrete logarithm of some `Z` to the base `Y`.
+This proof can be constructed by using *equality composition* on two Sigma protocols proving knowledge of the discrete logarithm between both pairs of points.
+In other words the prover proves knowledge of `a` such that `X = a * G` and `b` such that `Z = b * Y` and that `a = b`.
+We make the resulting Sigma protocol non-interactive by applying the Fiat-Shamir transformation with SHA256 as the challenge hash.
+For background on Sigma protocols and making them non-interactive see [[Sch]].
+
+To disambiguate the proof we first *tag* a SHA256 as specified in [[BIP340]] with a string that identifies the kind of statement we are proving: `tag_string = "eq(DLG(secp256k1),DL(secp256k1))"`.
```

Comment:
is one `DLG` and the other `DL` in the tag intentionally?

Reply:
Yes. The idea I am working towards is that this string should fully describe all the elements that follow in the hash.
DLG implies that `G` is being used and therefore we don't bother hashing it as part of the statement.
So following `eq(DLG(secp256k1),DL(secp256k1))` you have three points `X, Y, Z` (G is implicit) but following  `eq(DL(secp256k1),DL(secp256k1))` you would have four points `W,X,Y,Z`.